lib/ed25519.js

```diff
@@ -36,6 +44,132 @@ ed25519.constants.SEED_BYTE_LENGTH = 32;
 ed25519.constants.SIGN_BYTE_LENGTH = 64;
 ed25519.constants.HASH_BYTE_LENGTH = 64;
 
+/**
+ * ASN1 public key validator for importing ed25519 keys in DER format.
+ * @type {Object}
+*/
+var ed25519PublicKeyValidator = forge.ed25519.publicKeyValidator = {
+  name: 'ed25519PublicKey',
+  tagClass: asn1.Class.UNIVERSAL,
+  type: asn1.Type.SEQUENCE,
+  constructed: true,
+  composed: true,
+  value: [{
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.SEQUENCE,
+    constructed: true,
+    composed: true,
+    value: [{
+      tagClass: asn1.Class.UNIVERSAL,
+      type: asn1.Type.OID,
+      constructed: false,
+      composed: false,
+      capture: 'ed25519Oid'
+    }]
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.BITSTRING,
+    constructed: false,
+    composed: true,
+    captureBitStringValue: 'ed25519PublicKey'
+  }]
+};
+
+/**
+ * ASN1 private key validator for importing ed25519 keys in DER format.
+ * @type {Object}
```

Comment:
Please add a note here:

```
This validator is for a `OneAsymmetricKey` ASN.1 type as specified in RFC5958 (https://tools.ietf.org/html/rfc5958). This type has the same format as a simple `PrivateKeyInfo`
for which there is already a validator specified in forge's `rsa` module.

TODO: Reuse `PrivateKeyInfo` validator from forge `rsa`.
```

Reply:
If helpful, we can include the link to the other validator, but it could get out of date.

https://github.com/digitalbazaar/forge/blob/master/lib/rsa.js#L96